### ISSUE TITLE
NVMe Over Fabric get error

### ISSUE THREAD

**Issue by cocalele** [zrlio/softiwarp#11]
While testing NoF with softiwarp, the initiator get following error
```
# nvme discover -t rdma -a 127.0.0.1 -s 1023
Failed to write to /dev/nvme-fabrics: Connection refused

# dmesg
nvme nvme0: rdma_resolve_addr wait failed (-111).
```

**Comment by cocalele** [comment]
The reason is becauses `ofa_dev->alloc_mr` is not implemented in function `siw_device_create`.

But I have no idea on how to implement this function :(

**Comment by BernardMetzler** [comment]
right. will look into it. have to understand & enable it for newer kernels.

**Comment by cocalele** [comment]
`map_mr_sg` is also require by NoF but not implemented.

**Comment by BernardMetzler** [comment]
I opened a development branch 'dev-siw.mem_ext'. It implements mentioned memory management extensions. I tested with NVMeF (both initiator and target). Please treat that code as purely experimental - stability and performance will improve over time. It runs on kernels 4.8 and 4.10.
